### PR TITLE
Fix: Area Miniboss Mob Detection

### PR DESCRIPTION
**Pull request by ItsEmpa**
## What
Migrates Area Miniboss features to use mob detection instead, and fixes an issue with formatting Durations with milliseconds.

## Changelog Fixes
+ Fixed durations displaying with milliseconds. - Empa

## Changelog Technical Details
+ Refactored Area Miniboss Features to use mob detection. - Empa

### CODE REVIEW

**Review by hannibal002 — CHANGES_REQUESTED** [review]
Now the time looks wrong for area mobs. I'm sure it was not like that
image

Moreover, the new mob detection doesn't detect mobs when you first enter a region/"run into them". It only detects area mobs when they spawn around you. The old system detects them on first join as well.

**Comment by ItsEmpa** [comment]
timer issue is fixed now.
and as i said in discord, the reason the minibosses arent detected outside is because we now check that you are in a slayer area. i can change this if really needed

File: src/main/java/at/hannibal2/skyhanni/features/combat/mobs/AreaMiniBossFeatures.kt [hunk 1]
```diff
@@ -34,45 +24,31 @@ object AreaMiniBossFeatures {
...
-
-            if (config.areaBossHighlight) {
-                val color = bossType.color.toColor().withAlpha(bossType.colorOpacity)
-                RenderLivingEntityHelper.setEntityColorWithNoHurtTime(entity, color)
-                { config.areaBossHighlight && SlayerAPI.isInAnyArea }
-            }
-
-            // TODO add sound
+    fun onMobSpawn(event: MobEvent.Spawn.SkyblockMob) {
+        if (!SlayerAPI.isInAnyArea) return
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/combat/mobs/AreaMiniBossFeatures.kt]: waiting for deletion (as discussed in discord)

**Review by hannibal002 — CHANGES_REQUESTED** [review]
beep boop

**Review by hannibal002 — APPROVED** [review]
Works for me

**Pull request merged**